cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -430,9 +430,14 @@ private DatasetTypeMeta getTypeInfo(NamespaceId namespaceId, String typeName) th
     try {
       return typeService.getType(datasetTypeId);
     } catch (DatasetTypeNotFoundException e) {
-      // Type not found in the instance's namespace. Now try finding it in the system namespace
-      DatasetTypeId systemDatasetTypeId = ConversionHelpers.toDatasetTypeId(NamespaceId.SYSTEM, typeName);
-      return typeService.getType(systemDatasetTypeId);
+      try {
+        // Type not found in the instance's namespace. Now try finding it in the system namespace
+        DatasetTypeId systemDatasetTypeId = ConversionHelpers.toDatasetTypeId(NamespaceId.SYSTEM, typeName);
+        return typeService.getType(systemDatasetTypeId);
+      } catch (DatasetTypeNotFoundException e1) {
```

Comment:
If the dataset type - say myDS - does not exist in the requested namespace - say myNS -  we try to see whether it exists in the system namespace. If that also fails, then we should throw the exception with the original namespace. Previously, we would throw an exception saying "Dataset Type system.myDS" does not exist. But we should throw "Dataset Type myNS.myDS does not exist".